lib/chef/provider/package.rb

```diff
@@ -500,11 +500,13 @@ def target_version_array
                 end
 
               when :install
-
                 if new_version
                   if version_requirement_satisfied?(current_version, new_version)
                     logger.trace("#{new_resource} #{package_name} #{current_version} satisifies #{new_version} requirement")
                     target_version_array.push(nil)
+                  elsif current_version && version_compare(current_version, new_version) == 1 && !allow_downgrade
```

Comment:
Check `allow_downgrade` before `version_compare` since that's a cheaper condition.

Reply:
Nice pick thanks a lot @coderanger